### PR TITLE
[exporterhelper] Adding shutdown & startup debug logs for retry, queue, and batch sender

### LINKED ISSUE #10201: There should be more observability around exporter helper start up and shutdown
## Important (read before submitting)

**Is your feature request related to a problem? Please describe.**
As a user of otel collector, I find it hard to see which exporter helper component is stuck starting up or shutting down. Some of the examples in the past:
- persistent queue helper start up can be stuck: https://github.com/open-telemetry/opentelemetry-collector/issues/9451
- batch sender shut down can be stuck due to config: https://github.com/open-telemetry/opentelemetry-collector/issues/10061

When debugging these problems, it isn't clear that the components are stuck.

**Describe the solution you'd like**
A debug message is added at the beginning of startup/shutdown and at the end of the startup/shutdown. Components we should add:
- retry sender
- batch sender
- queue sender

**Describe alternatives you've considered**
Add it as INFO logs but it can be too noisy

**Additional context**

### PR DESCRIPTION
**Pull request by timannguyen**
#### Description

Adding debug logs to starting and shutdown of batch, retry, and queue sender in exporterhelper to understand which component is stuck shutting down or starting up. This is added in order to improve debugging when one of these components are stuck shutting down or starting up.

#### Link to tracking issue
Fixes #10201

#### Testing

N/A

#### Documentation

N/A

### CODE REVIEW

**Review by splunkericl — APPROVED** [review]

File: exporter/exporterhelper/queue_sender.go [hunk 1]
```diff
@@ -137,15 +139,20 @@ func (qs *queueSender) Start(ctx context.Context, host component.Host) error {
 			return nil
 		}))
 
+	qs.logger.Debug("Started Queue Sender", zap.String(obsmetrics.ExporterKey, qs.fullName), zap.Error(err))
 	errs = multierr.Append(errs, err)
```
Comment (atoulme) [inline exporter/exporterhelper/queue_sender.go:143]:
```suggestion
	errs = multierr.Append(errs, err)
	qs.logger.Debug("Started Queue Sender", zap.String(obsmetrics.ExporterKey, qs.fullName), zap.Error(errs))
```

File: exporter/exporterhelper/batch_sender.go [hunk 2]
```diff
@@ -91,6 +94,7 @@ func (bs *batchSender) Start(_ context.Context, _ component.Host) error {
 		}
 	}()
 
+	bs.logger.Debug("Started Batch Sender", zap.String(obsmetrics.ExporterKey, bs.fullName))
```
Comment (bogdandrutu) [inline exporter/exporterhelper/batch_sender.go:97]: I think the `obsmetrics.ExporterKey` entry is not necessary, the logger will have a "name" entry (see https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/internal/components/loggers.go#L38) with this.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]

File: exporter/exporterhelper/batch_sender.go (hunk 2 above)
Comment (timannguyen) [reply exporter/exporterhelper/batch_sender.go]: removed the exporter key from all the logs.

File: exporter/exporterhelper/queue_sender.go (hunk 1 above)
Comment (timannguyen) [reply exporter/exporterhelper/queue_sender.go]: moved

**Comment by timannguyen** [comment]
@bogdandrutu is there anything else i need to address?

**Pull request closed without merging**